app/js/demo-controller.js

```diff
@@ -60,9 +70,28 @@ export class DemoPaywallController {
         if (this.completeDeferredAccountCreation_(entitlements)) {
           // Do nothing.
         } else {
-          // Entitlements available: open access.
-          this.openPaywall_();
-          entitlements.ack();
+          const account = new Account('John Doe', 'johndoe@email.com');
+          const accountPromise = new Promise(resolve => {
+            setTimeout(() => {
+              resolve(account);
+            }, 5000);
+          });
+          this.subscriptions.waitForSubscriptionLookup(accountPromise)
+              .then(account => {
+                if (account) {
+                  this.showLoginPromptOrNotification_(this.subscriptions,
+                      this.consentRequired_).then(result => {
+                        if (result) {
+                          this.openPaywall_();
+                          entitlements.ack();
+                        }
+                      }, reason => {
```

Comment:
It doesn't look like catch clause here doing anything. Remove?

Reply:
done